### PR TITLE
Update subcommands

### PR DESCRIPTION
**Pull request by katiedavis**
Updates generate to use subcommand pattern

Adds the ability to bi-pass selection of page type by passing a `TYPE` flag.

### CODE REVIEW

File: lib/shopify-cli/commands/generate.rb [hunk 1]
```diff
@@ -66,8 +56,8 @@ def self.extended_help
         HELP
       end
 
-      def self.run_generate(script, name, ctx)
-        stat = ctx.system(script)
+      def self.run_generate(script, name)
+        stat = @ctx.system(script)
```
Comment (tylerball) [inline lib/shopify-cli/commands/generate.rb]: I don't think this will work because its a class method, which is why it was passed in

Comment (katiedavis) [reply lib/shopify-cli/commands/generate.rb]: I know you're not wrong, but it does work. Why would it work? It totally runs the script to generate all the commands.

Comment (tanema) [reply lib/shopify-cli/commands/generate.rb]: I am worried about this. If we don't know how it works, we should probably not use it. Why did you change it?

Comment (katiedavis) [reply lib/shopify-cli/commands/generate.rb]: Hey! I changed it back, but I do really want to know why it works.

**Pull request merged**